### PR TITLE
Store configuration struct in status

### PR DESCRIPTION
**Pull request by coreydaley**
- Store the configuration struct in the status
 - Move Status.ManagedStorage to Status.Storage.Managed

The pertinent part of the image registry resource ends up looking like this:
```
   storage:
      managed: true
      state:
        s3:
          bucket: image-registry-33d2533f-f588-11e8-a437-0a580a80020b
          region: us-west-1
```

### CODE REVIEW

**Comment by coreydaley** [comment]
@openshift/sig-developer-experience ptal

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func newDriver(crname string, crnamespace string, cfg *opapi.ImageRegistryConfig
 	}
 
 	if cfg.Filesystem != nil && cfg.Filesystem.VolumeSource.EmptyDir != nil {
-		drivers = append(drivers, emptydir.NewDriver(crname, crnamespace, nil))
+		drivers = append(drivers, emptydir.NewDriver(crname, crnamespace, cfg.Filesystem))
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply pkg/storage/storage.go]: updated in current push

File: pkg/storage/storage.go [hunk 2]
```diff
@@ -83,11 +83,7 @@ func NewDriver(crname string, crnamespace string, cfg *opapi.ImageRegistryConfig
 		case clusterconfig.StorageTypeFileSystem:
 			cfg.Filesystem = &opapi.ImageRegistryConfigStorageFilesystem{}
 		case clusterconfig.StorageTypeEmptyDir:
-			cfg.Filesystem = &opapi.ImageRegistryConfigStorageFilesystem{
-				VolumeSource: corev1.VolumeSource{
-					EmptyDir: &corev1.EmptyDirVolumeSource{},
-				},
-			}
+			cfg.Filesystem = &opapi.ImageRegistryConfigStorageFilesystem{}
```
[the remark this answers is not in the document]
Comment (coreydaley) [reply pkg/storage/storage.go]: updated in current push

File: pkg/storage/emptydir/emptydir.go [hunk 3]
```diff
@@ -52,5 +52,11 @@ func (d *driver) Volumes() ([]corev1.Volume, []corev1.VolumeMount, error) {
 }
 
 func (d *driver) CompleteConfiguration(customResourceStatus *opapi.ImageRegistryStatus) error {
+
+	d.Config.VolumeSource = corev1.VolumeSource{
```
Comment (dmage) [inline pkg/storage/emptydir/emptydir.go]: Is it an intention to modify the custom resource? Why do we need it?

Comment (coreydaley) [reply pkg/storage/emptydir/emptydir.go]: Updated, that was leftover from a previous change that got updated.

**Comment by bparees** [comment]
> Why do we need to copy the entire storage config to status?

i haven't gone through the fields line by line, but in general the idea is that if something about the storage configuration changes, we can recognize it by comparing the spec and status copies, and then apply the changes to the registry, so we need the full copy to compare.  If you think there are specific fields we can do without, we should definitely discuss them.

**Pull request merged**